File: python/week04/score_stats.py
def find_total(scores):
    total = 0

    for score in scores:
        total = total + score

    return total


def find_average(scores):
    total = find_total(scores)
    average = total / len(scores)

    return average


def find_max(scores):
    max_score = scores[0]

    for score in scores:
        if score > max_score:
            max_score = score

    return max_score


def find_min(scores):
    min_score = scores[0]

    for score in scores:
        if score < min_score:
            min_score = score

    return min_score


def count_pass(scores):
    pass_count = 0

    for score in scores:
        if score >= 60:
            pass_count = pass_count + 1

    return pass_count


def count_fail(scores):
    fail_count = 0

    for score in scores:
        if score < 60:
            fail_count = fail_count + 1

    return fail_count


scores = [88, 59, 76, 92, 45, 100, 67]

total = find_total(scores)
average = find_average(scores)
max_score = find_max(scores)
min_score = find_min(scores)
pass_count = count_pass(scores)
fail_count = count_fail(scores)

print("total =", total)
print("average =", average)
print("max =", max_score)
print("min =", min_score)
print("pass count =", pass_count)
print("fail count =", fail_count)

File: java/week04/ScoreStats.java
public class ScoreStats {

    public static void main(String[] args) {

        int[] scores = {88, 59, 76, 92, 45, 100, 67};

        int total = findTotal(scores);
        double average = findAverage(scores);
        int max = findMax(scores);
        int min = findMin(scores);
        int passCount = countPass(scores);
        int failCount = countFail(scores);

        System.out.println("total = " + total);
        System.out.println("average = " + average);
        System.out.println("max = " + max);
        System.out.println("min = " + min);
        System.out.println("pass count = " + passCount);
        System.out.println("fail count = " + failCount);
    }

    public static int findTotal(int[] scores) {
        int total = 0;

        for (int score : scores) {
            total = total + score;
        }

        return total;
    }

    public static double findAverage(int[] scores) {
        int total = findTotal(scores);
        return (double) total / scores.length;
    }

    public static int findMax(int[] scores) {
        int max = scores[0];

        for (int score : scores) {
            if (score > max) {
                max = score;
            }
        }

        return max;
    }

    public static int findMin(int[] scores) {
        int min = scores[0];

        for (int score : scores) {
            if (score < min) {
                min = score;
            }
        }

        return min;
    }

    public static int countPass(int[] scores) {
        int count = 0;

        for (int score : scores) {
            if (score >= 60) {
                count = count + 1;
            }
        }

        return count;
    }

    public static int countFail(int[] scores) {
        int count = 0;

        for (int score : scores) {
            if (score < 60) {
                count = count + 1;
            }
        }

        return count;
    }
}